src/containers/Tenant/Diagnostics/TopicData/TopicDataControls/TopicDataControls.tsx

```diff
@@ -0,0 +1,172 @@
+
+    return (
+        <React.Fragment>
+            <Select
+                className={b('partition-select')}
+                label={i18n('label_partition-id')}
+                options={partitionsToSelect}
+                value={selectedPartition ? [selectedPartition] : undefined}
+                onUpdate={handleSelectedPartitionChange}
+                filterable={partitions && partitions.length > 5}
+                disabled={!partitions || !partitions.length}
+                errorPlacement="inside"
+                errorMessage={prepareErrorMessage(partitionsError)}
+                error={Boolean(partitionsError)}
+                loading={partitionsLoading}
+            />
+            <TopicDataStartControls />
+            <TableColumnSetup
+                popupWidth={242}
+                items={columnsToSelect}
+                showStatus
+                onUpdate={handleSelectedColumnsUpdate}
+                sortable={false}
+            />
+            <EntitiesCount
+                label={i18n('label_offset')}
+                current={`${formatNumber(initialOffset)}—${formatNumber(endOffset)}`}
+            />
+        </React.Fragment>
+    );
+}
+
+function TopicDataStartControls() {
+    const {
+        selectedOffset,
+        startTimestamp,
+        topicDataFilter,
+        handleSelectedOffsetChange,
+        handleStartTimestampChange,
+        handleTopicDataFilterChange,
+    } = useTopicDataQueryParams();
+
+    const onFilterChange = React.useCallback(
+        (value: TopicDataFilterValue) => {
+            handleTopicDataFilterChange(value);
+        },
+        [handleTopicDataFilterChange],
+    );
+    const onStartOffsetChange = React.useCallback(
+        (value: string) => {
+            handleSelectedOffsetChange(convertToNumber(value));
+        },
+        [handleSelectedOffsetChange],
+    );
+    const handleFromTimestampChange = React.useCallback(
+        (data: Value | null) => {
+            let newValue: number | undefined;
+            if (data) {
+                const {value, type} = data;
+
+                if (type === 'absolute') {
+                    newValue = value.valueOf();
+                } else if (type === 'relative') {
+                    const absoluteValue = dateTimeParse(value);
+                    newValue = absoluteValue ? absoluteValue.valueOf() : undefined;
+                }
+            }
+            handleStartTimestampChange(newValue);
+        },
+        [handleStartTimestampChange],
+    );
+
+    const startDateTime = dateTimeParse(Number(startTimestamp));
+
+    return (
+        <React.Fragment>
+            <RadioButton<TopicDataFilterValue> value={topicDataFilter} onUpdate={onFilterChange}>
+                <RadioButton.Option value="TIMESTAMP">
+                    {TopicDataFilterValues.TIMESTAMP}
+                </RadioButton.Option>
+                <RadioButton.Option value="OFFSET">
+                    {TopicDataFilterValues.OFFSET}
```

Comment:
I suppose value in RadioButton should be TopicDataFilterValues.OFFSET and text i18(...)

according to current project rules